modules/xsession.nix

```diff
@@ -212,6 +212,10 @@ in {
         while [ -n "$(systemctl --user --no-legend --state=deactivating list-units)" ]; do
           sleep 0.5
         done
+
+        ${optionalString (cfg.importedVariables != [ ])
+        ("systemctl --user unset-environment "
+          + toString (unique cfg.importedVariables))}
```

Comment:
```suggestion
          + escapeShellArgs (unique cfg.importedVariables))}
```

Reply:
I also don't think we need that `unique`. (Remember it has quadratic complexity.)

Reply:
This was copied from the same file above when it uses `import-environment`, so I'll apply the change to both places it's used.

Reply:
> I also don't think we need that `unique`. (Remember it has quadratic complexity.)

surprisingly, it is somewhat required:

```bash
:;  AAA=5 systemctl --user import-environment AAA AAA
Failed to import environment: Invalid environment assignments

:; systemctl --user unset-environment AAA AAA
Failed to set environment: Invalid environment variable names or assignments
```

Reply:
Then maybe we can at least share the `unique` call. (Maybe even move it to an `apply` in the option?)

Reply:
`apply` seems sane to me, done!

(if you're really worried about performance that much, something dumb like `uniqueStrings = flip pipe [ (flip genAttrs id) attrNames ]` is also an option)